api/src/main/java/io/opentelemetry/metrics/LongGauge.java

```diff
@@ -53,6 +53,17 @@
 @ThreadSafe
 public interface LongGauge extends Gauge<BoundLongGauge> {
 
+  /**
+   * Sets the given value for the gauge.
+   *
+   * <p>The value added is associated with the current {@code Context}.
+   *
+   * @param val the new value.
+   * @param labelSet the labels to be associated to this value
+   * @since 0.1.0
+   */
+  void set(long val, LabelSet labelSet);
```

Comment:
How much would you hate me if I say that I like more:
`void set(LableSet labelSet, long value)`?

Reply:
heh. no hate here. I'm not opposed to that. Can you explain why it makes more sense to you the other way around?

Reply:
I feel like reading:
"Add for LabelSet this value", also kind of matches other APIs like prom/micrometer you specify the labels before the value. Also matches more instrument.bind(ls).add().

Reply:
I can see that. My reasoning was that the most "important" thing was the value, and the labels are secondary information to that. I'd love a tie-breaker opinion. @tylerbenson ? @breedx-nr ?

Reply:
Let's keep it as you have because we may want to add overloads with no LS for example and that will be better. I am fine keeping it this way

Reply:
👍